pkg/service/backup/worker_schema.go

```diff
@@ -100,3 +125,51 @@ func (w *worker) UploadSchema(ctx context.Context, hosts []hostInfo) (stepError
 
 	return hostsInParallel(hostPerLocation, parallel.NoLimit, f, notify)
 }
+
+func marshalAndCompressSchema(schema query.DescribedSchema) (*bytes.Buffer, error) {
+	rawSchema, err := json.Marshal(schema)
+	if err != nil {
+		return nil, errors.Wrap(err, "marshal schema")
+	}
+
+	b := &bytes.Buffer{}
```

Comment:
I think that it depends on the context.
Here, we return `*bytes.Buffer{}` from this function, so it can't be initialized on this function stack, as it needs to be available from outside of this function body. So I doubt that changing it to `var b bytes.Buffer` would accomplish anything (unless  I'm missing something).

This could work if both reading and uploading schema happened in the same function body. This could be done by combining the `StageAwaitSchema` and `StageSchema` stages - I don't see why they need to be separate. Nevertheless, this seems more like a refactoring task that is outside of the scope of this PR and can be included in another one.

Reply:
> Here, we return *bytes.Buffer{} from this function, so it can't be initialized on this function stack, as it needs to be available from outside of this function body. So I doubt that changing it to var b bytes.Buffer would accomplish anything (unless I'm missing something).

Makes sense, but please be explicit with the type that the function returns.
Use bytes.Buffer as a return type instead of the pointer.